### PR TITLE
Fixed a crash when exits were determined from dead buildings

### LINKED ISSUE #19707: A crash related with Exit trait (and probably plus AI)
See [issue](https://github.com/ABrandau/Shattered-Paradise-SDK/issues/42)

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/19707

### CODE REVIEW

**Review by reaperrr — DISMISSED** [review]
Untested, but makes sense

**Comment by dnqbob** [comment]
going to test it

File: OpenRA.Mods.Common/Traits/Buildings/Exit.cs [hunk 1]
```diff
@@ -35,12 +35,12 @@ public class ExitInfo : ConditionalTraitInfo, Requires<IOccupySpaceInfo>
 		[Desc("Exits with larger priorities will be used before lower priorities.")]
 		public readonly int Priority = 1;
 
-		public override object Create(ActorInitializer init) { return new Exit(init, this); }
+		public override object Create(ActorInitializer init) { return new Exit(this); }
 	}
 
 	public class Exit : ConditionalTrait<ExitInfo>
 	{
-		public Exit(ActorInitializer init, ExitInfo info)
```
Comment (dnqbob) [inline OpenRA.Mods.Common/Traits/Buildings/Exit.cs:43]: I don't know if this fix is necessary

**Comment by dnqbob** [comment]
A sad thing is, I cannot reproduce it in replay. The replay itself is buggy due to game speed is too quick.

File: OpenRA.Mods.Common/Traits/Buildings/Exit.cs (hunk 1 above)
Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/Buildings/Exit.cs:43]: by drop the actor initializer

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Buildings/Exit.cs:43]: This is not a fix, just a cleanup while editing the same file.

Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/Buildings/Exit.cs:43]: Got it

File: OpenRA.Mods.Common/Traits/Buildings/Exit.cs [hunk 2]
```diff
@@ -62,6 +62,9 @@ public static Exit NearestExitOrDefault(this Actor actor, WPos pos, string produ
 
 		public static IEnumerable<Exit> Exits(this Actor actor, string productionType = null)
 		{
+			if (actor.IsDead)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Buildings/Exit.cs]: `if (!actor.IsInWorld || actor.Disposed)` might be a better fit?

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Buildings/Exit.cs]: I don't know the exact difference, but that seems to match the stacktrace better.

**Review by RoosterDragon — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e39eb80ced15a1294922d275ccb361e7420e299d...e2cd661c65f83bf92b0e46a5eb5c50061a8c4d01)

**Pull request merged**